### PR TITLE
fix(cmd): replace creack/pty with go-pty for cross-platform PTY support

### PR DESCRIPTION
**Pull request by thejoeejoee**
## Summary

- Replace `creack/pty/v2` with `github.com/aymanbagabas/go-pty` v0.2.3 for cross-platform PTY (Unix + Windows ConPTY)
- Fix tiny-window race: resize PTY *before* starting child process (`New → Resize → Start`)
- Add Windows resize polling (250ms interval, no SIGWINCH on Windows)
- Add platform-specific `terminateSignal` (SIGTERM on Unix, os.Kill on Windows)

### CODE REVIEW

File: internal/cmd/attach.go [hunk 1]
```diff
@@ -138,31 +137,44 @@ func attachOnHost(ctx context.Context, ws *workspace.Resolved, dir string, passw
...
+	// default PTY dimensions at startup.
+	fd := int(os.Stdin.Fd()) //nolint:gosec // Fd() fits in int on all supported platforms
+	if term.IsTerminal(fd) {
+		if w, h, sizeErr := term.GetSize(fd); sizeErr == nil {
+			_ = p.Resize(w, h)
+		}
+	}
+
+	cmd := p.Command(binary, args...)
+	cmd.Env = env
```
[the remark this answers is not in the document]
Comment (thejoeejoee) [reply internal/cmd/attach.go:205]: go-pty unconditionally sets `cmd.Stderr = pty.slave` internally (see `cmd_unix.go`), so there's no way to redirect stderr separately from the PTY stream. The previous code with creack/pty had the same effective behavior — `cmd.Stderr = os.Stderr` was overridden by creack's `StartWithAttrs` which connects all stdio to the PTY slave. This is inherent PTY behavior (stderr and stdout merge on the slave side), not a regression.

File: internal/docker/docker.go [hunk 2]
```diff
@@ -306,6 +308,113 @@ func (c *Client) Exec(ctx context.Context, args []string, env []string, stdin io
...
+	Width  uint
+	Height uint
+}
+
+// ExecInteractive runs a command in the opencode container using the Docker
+// Engine API directly (not Compose SDK). This gives the caller full control
+// over TTY resize via the resizeCh channel. Each TerminalSize sent on resizeCh
+// triggers a ContainerExecResize call. The method blocks until the exec
+// process exits or the context is cancelled.
+func (c *Client) ExecInteractive(ctx context.Context, containerID string, args []string, env []string, stdin io.Reader, stdout io.Writer, resizeCh <-chan TerminalSize) error {
```
[the remark this answers is not in the document]
Comment (thejoeejoee) [reply internal/docker/docker.go:321]: This method exercises ContainerExecCreate/Attach/Resize which require a running Docker daemon — unit-testing it without heavy mocking would be artificial. The project's test strategy uses real Docker only in integration tests (`//go:build integration` in `internal/integration_test.go`). The exec path is covered there end-to-end. Adding unit tests that mock the full Engine API would duplicate what integration tests already validate with higher fidelity.
